daemon/src/db/failed.rs

```diff
@@ -0,0 +1,556 @@
+//! This module allows us to move failed CFDs to a separate table so
+//! that we can reason about loading them independently from open
+//! CFDs.
+//!
+//! Failed CFDs are ones that did not reach the point of having a DLC
+//! on chain. This can happen either because the taker's order was
+//! rejected or contract setup did not finish successfully.
+//!
+//! Therefore, it also provides an interface to load failed CFDs: the
+//! `FailedCfdAggregate` trait. Implementers of the trait will be able
+//! to call the `crate::db::load_all_cfds` API, which loads all types
+//! of CFD.
+
+use crate::db;
+use crate::db::delete_from_cfds_table;
+use crate::db::delete_from_events_table;
+use crate::db::event_log::EventLog;
+use crate::db::event_log::EventLogEntry;
+use crate::db::load_cfd_events;
+use crate::db::load_cfd_row;
+use crate::db::CfdAggregate;
+use crate::db::Connection;
+use anyhow::bail;
+use anyhow::Result;
+use model::impl_sqlx_type_display_from_str;
+use model::long_and_short_leverage;
+use model::Contracts;
+use model::EventKind;
+use model::FeeAccount;
+use model::Fees;
+use model::FundingFee;
+use model::Identity;
+use model::Leverage;
+use model::OrderId;
+use model::Position;
+use model::Price;
+use model::Role;
+use model::Timestamp;
+use sqlx::pool::PoolConnection;
+use sqlx::Connection as _;
+use sqlx::Sqlite;
+use sqlx::Transaction;
+use std::fmt;
+use std::str;
+
+/// A trait for building an aggregate based on a `FailedCfd`.
+pub trait FailedCfdAggregate: CfdAggregate {
+    fn new_failed(args: Self::CtorArgs, cfd: FailedCfd) -> Self;
+}
+
+/// Data loaded from the database about a failed CFD.
+#[derive(Debug, Clone, Copy)]
+pub struct FailedCfd {
+    pub id: OrderId,
+    pub position: Position,
+    pub initial_price: Price,
+    pub taker_leverage: Leverage,
+    pub n_contracts: Contracts,
+    pub counterparty_network_identity: Identity,
+    pub role: Role,
+    pub fees: Fees,
```

Comment:
I remember we discussed this: The `fees` are added because we display them in the UI?
Maybe we should actually re-consider this. It is somewhat confusing and should always be `0`. I think the UI is also confusing for displaying it.

We don't have to block this PR by this change, can do in a follow-up.

Reply:
Fair point! I just wanted to keep the behaviour the same as it currently is in production for the time being.